lldb/source/Plugins/ExpressionParser/Swift/SwiftREPL.cpp

```diff
@@ -551,10 +551,9 @@ int SwiftREPL::CompleteCode(const std::string &current_code,
 
     auto *target_swift_ast =
         llvm::dyn_cast_or_null<SwiftASTContext>(&*type_system_or_err);
-    if (target_swift_ast)
-      m_swift_ast_sp.reset(new SwiftASTContext(*target_swift_ast));
+    m_swift_ast = target_swift_ast;
   }
-  SwiftASTContext *swift_ast = m_swift_ast_sp.get();
+  SwiftASTContext *swift_ast = m_swift_ast;
```

Comment:
I think we should get rid of m_swift_ast and only use the local copy, because it's not safe to hold on to the pointer without also holding on to the lock.